src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -515,45 +524,81 @@ def _find_routine(call_node):
             for routine in call_node.root.children:
                 if isinstance(routine, Routine) and routine.name == name:
                     return routine
+            else:
+                raise TransformationError(
+                    f"Failed to find the source code of the unresolved "
+                    f"routine '{routine.name}' after trying wildcard imports "
+                    f"'{wildcard_names}' and all routines that are not in "
+                    f"containers.")
 
         elif routine_sym.is_import:
             container_symbol = routine_sym.interface.container_symbol
             routine = InlineTrans._find_routine_in_container(
                 call_node, container_symbol)
             if routine:
                 return routine
-
-        raise TransformationError(
-            f"Failed to find the source for routine '{name}' and "
-            f"therefore cannot inline it.")
+            raise TransformationError(
+                f"Failed to find the source for the imported routine "
```

Comment:
"...for routine '{...}' imported from '{container_symbol.name}' and therefore..." perhaps?

Reply:
OK.